android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/manager/FetchDownloadManager.kt

```diff
@@ -128,9 +128,9 @@ class FetchDownloadManager<T : AbstractFetchDownloadService>(
         val downloadID = intent.getStringExtra(EXTRA_DOWNLOAD_ID) ?: ""
         val download = store.state.downloads[downloadID]
         val downloadStatus = intent.getSerializableExtraCompat(EXTRA_DOWNLOAD_STATUS, Status::class.java)
-            as Status
+            as Status?
```

Comment:
@Amejia481 I'm here after checking the reports in this [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1877881). I have limited info so please bear with me, quickly checking the [crash log](https://mozilla.sentry.io/issues/4447065477/?project=6375561&query=is%3Aunresolved+FetchDownloadManager&referrer=issue-stream), it seems we are [unsafe casting](https://kotlinlang.org/docs/typecasts.html#unsafe-cast-operator) using `as` to a nullable type here. I think we want a [safe case](https://kotlinlang.org/docs/typecasts.html#safe-nullable-cast-operator) which doesn't crash when it's null with `as?`. 

wdyt?